core/src/main/java/org/infinispan/xsite/irac/IracResponseCollector.java

```diff
@@ -1,136 +1,111 @@
 
-   public IracResponseCollector(String cacheName, Collection<IracManagerKeyState> batch, IracResponseCompleted listener) {
+   public IracResponseCollector(String cacheName, IracXSiteBackup backup, Collection<IracManagerKeyState> batch, IracResponseCompleted listener) {
       this.cacheName = cacheName;
+      this.backup = backup;
       this.batch = batch;
       this.listener = listener;
-      countDownRunnable = new CountDownRunnable(this);
-      failedKeys = IntSets.mutableEmptySet(batch.size());
-   }
-
-   public void dependsOn(IracXSiteBackup backup, CompletionStage<? extends IntSet> request) {
-      countDownRunnable.increment();
-      request.whenComplete((bitSet, throwable) -> onResponse(backup, bitSet, throwable));
-   }
-
-   public CompletionStage<Void> freeze() {
-      countDownRunnable.freeze();
-      return completableFuture;
+      failedKeys = IntSets.concurrentSet(batch.size());
    }
 
-   private void onResponse(IracXSiteBackup backup, IntSet rspIntSet, Throwable throwable) {
+   @Override
+   public void accept(IntSet rspIntSet, Throwable throwable) {
       boolean trace = log.isTraceEnabled();
-      try {
-         if (throwable != null) {
-            exceptionReceived = true;
-            if (DefaultTakeOfflineManager.isCommunicationError(throwable)) {
-               //in case of communication error, we need to back-off.
-               RESULT_UPDATED.set(this, IracBatchSendResult.BACK_OFF_AND_RETRY);
-            } else if (result == IracBatchSendResult.OK) {
-               //don't overwrite communication errors
-               RESULT_UPDATED.compareAndSet(this, IracBatchSendResult.OK, IracBatchSendResult.RETRY);
-            }
-            if (backup.logExceptions()) {
-               log.warnXsiteBackupFailed(cacheName, backup.getSiteName(), throwable);
-            } else if (trace) {
-               log.tracef(throwable, "[IRAC] Encountered issues while backing up data for cache %s to site %s", cacheName, backup.getSiteName());
-            }
-         } else {
-            if (trace) {
-               log.tracef("[IRAC] Received response from site %s (%d missing): %s", backup.getSiteName(), countDownRunnable.missing(), rspIntSet);
-            }
-            mergeIntSetResult(rspIntSet);
-            // if some keys failed to apply, we need to retry.
-            if (!rspIntSet.isEmpty() && result == IracBatchSendResult.OK) {
-               RESULT_UPDATED.compareAndSet(this, IracBatchSendResult.OK, IracBatchSendResult.RETRY);
-            }
+      if (throwable != null) {
+         if (DefaultTakeOfflineManager.isCommunicationError(throwable)) {
+            //in case of communication error, we need to back-off.
+            backup.enableBackOffMode();
+            result = IracBatchSendResult.BACK_OFF_AND_RETRY;
+         } else if (result == IracBatchSendResult.OK) {
+            //don't overwrite communication errors
```

Comment:
as mentioned in the previous comment, you can invoke `backup.disableBackOffMode();` here.